### PR TITLE
[Enhancement] add soft link for the compile_commands.json

### LINKED ISSUE #21657: [Enhancement] automatically create soft link for command compile.json for clangd user
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Formerly, developers should manually set the `--compile-commands-dir` argument for clangd. According to clangd's [official documentation](https://clangd.llvm.org/installation), we can make one soft link to the 'compile_commands.json' so that clangd would automatically detect it and then do all the semantic analytical workloads.

### Solution

1. add one new ignore item for `compile_commands.json` in the root path
2. add logic in the build.sh which can detect the build dir and make soft link for the `compile_commands.json`

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xiedeyantu**
## Proposed changes

Issue Number: close #21657

add soft link for the compile_commands.json

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xiedeyantu** [comment]
@ByteYue please review this pr.

**Comment by ByteYue** [comment]
> @ByteYue please review this pr.

Sorry for my late response. LGTM

**Review by ByteYue — APPROVED** [review]

**Comment by ByteYue** [comment]
Maybe you should consider add one new item in .gitignore file to ignore the soft link file

**Comment by xiedeyantu** [comment]
> Maybe you should consider add one new item in .gitignore file to ignore the soft link file

finished

**Comment by ByteYue** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.34 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17167648032 Bytes

**Review by ByteYue — APPROVED** [review]

File: build.sh [hunk 1]
```diff
@@ -703,4 +703,8 @@ if [[ -n "${DORIS_POST_BUILD_HOOK}" ]]; then
     eval "${DORIS_POST_BUILD_HOOK}"
 fi
 
+if [[ -f "${CMAKE_BUILD_DIR}/compile_commands.json" ]]; then
+    ln -s "${CMAKE_BUILD_DIR}/compile_commands.json" "${DORIS_HOME}/compile_commands.json"
+fi
+
```
Comment (adonis0147) [inline build.sh:709]: I don't think the path `${DORIS_HOME}/compile_commands.json` is appropriate. We have two build scripts, one is for `src` and another one is for `test`. These build scripts will generate different `compile_commands.json`. Linking the `compile_commands.json` to the root of the project would affects these build targets.

**Pull request closed without merging**